modules/account-lib/src/coin/dot/singletonRegistry.ts

```diff
@@ -0,0 +1,24 @@
+import { TypeRegistry } from '@substrate/txwrapper-core/lib/types';
+import { getRegistry } from '@substrate/txwrapper-polkadot';
+import { PolkadotSpecNameType } from '@bitgo/statics';
+import { Metadata } from '@polkadot/types/metadata/Metadata';
+import { Material } from './iface';
+
+export class SingletonRegistry {
+  private static instance: TypeRegistry;
+  private static material: Material;
+  private static metadata: Metadata;
+
+  static getInstance(material: Material): TypeRegistry {
```

Comment:
For test coverage purposes, let's add a quick unit test check for this new method:
if === material and if !== material

Reply:
Done